### PR TITLE
Update php version, phpunit version and fix related problems

### PR DESCRIPTION
**Pull request by juffalow**
I had several problems with new version of PHPUnit. After these changes, everything is working for me.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,15 @@
 language: php
 
 php:
-    - 5.3
-    - 5.4
-    - 5.5
-    - 5.6
-    - hhvm
+    - 7.2
```
Comment (peter279k) [inline .travis.yml:5]: The `php-7.1` is stable. 

I think we should add the `php-7.1` version in Travis CI build.

File: composer.json [hunk 2]
```diff
@@ -12,14 +12,14 @@
         }
     ],
     "require": {
-        "php": ">=5.0.0"
-    },
-    "require-dev": {
-        "phpunit/phpunit": "~4.0"
+        "php": "^7.2"
```
Comment (peter279k) [inline composer.json]: I think this should be `^7.1` definition in `composer.json`.

**Review by peter279k — CHANGES_REQUESTED** [review]
It looks good to me, and some stuffs I mention should be concerned.

**Comment by peter279k** [comment]
@juffalow, could you concern about my request changes if possible?

**Comment by juffalow** [comment]
I updated those things but now the tests are failing. I checked *PHP Supported Versions* and version `7.1`had active support until 1 Dec 2018. Is it necessary to include it?

**Comment by peter279k** [comment]
> I updated those things but now the tests are failing. I checked _PHP Supported Versions_ and version `7.1`had active support until 1 Dec 2018. Is it necessary to include it?

@juffalow, you're right. We can remove the `php-7.1` at this moment.

But I just notice that the `php-7.1` will have the security support until 1 Dec 2019.

**Comment by juffalow** [comment]
What do you suggest? Should I take back just the last commit, or solve the problems with `7.1`?

**Comment by peter279k** [comment]
> What do you suggest? Should I take back just the last commit, or solve the problems with `7.1`?

The `php-7.1` has been very soon to be unsupported. I think it's fine to drop `php-7.1` version.

Just take back to last commit I think.

Just notice that this is about the community review, the final decision should be up to this repo owner.

**Comment by thomasrockhu** [comment]
Closing out as a result of https://github.com/codecov/example-php/commit/688d9f970c63de17fcca4db6421f45a55b7a615a#diff-354f30a63fb0907d4ad57269548329e3. Feel free to reopen if this is still relevant.

**Pull request closed without merging**